File: java/Stack/easy/remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String S) {
        int sum=0;
        StringBuilder a=new StringBuilder();
        char[] s=S.toCharArray();
        for(int i=0;i<s.length;i++){
            if((s[i]=='('&&sum++>0)||(s[i]==')'&&--sum>0)){
                a.append(s[i]);
            }
        }
        return a.toString();
    }
}

File: cpp/Stack/easy/remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string S) {
        string ans = "";
        int opened = 0;
        for(auto i : S) {
            if(i == '(') {
                if(opened > 0) {
                    ans += i;
                }
                opened++;
            } else {
                if(opened > 1) {
                    ans += i;
                } 
                opened--;
            }
        }
        return ans;
    }
};
